react/src/components/NavBar/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+import 'bootstrap/dist/css/bootstrap.css';
+
+import UserAvatar from '../UserAvatar';
+import routes from '../../constants/routes'
+import wbooksLogoSvg from '../../assets/wbooks_logo.svg';
+import addBookSvg from '../../assets/add_book.svg';
+import notificationsSvg from '../../assets/notifications.svg';
+
+import './styles.css';
+
+class NavBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      dropdownOpen: false
+    };
+  }
+
+  toggle() {
+    this.setState({
+      dropdownOpen: !this.state.dropdownOpen
+    });
```

Comment:
Since this update depends on previous state, I'd use the alternative form of `setState`, passing a function:
```
this.setState(prevState => ({ dropdownOpen: !prevState.dropdownOpen }));
```

Reply:
Complementary material:
https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b